### PR TITLE
Collections: add books from a folder

### PR DESCRIPTION
**Pull request by hius07**
Kind of "flat" library.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12892)

### CODE REVIEW

File: frontend/apps/filemanager/filemanagercollection.lua [hunk 1]
```diff
@@ -239,6 +240,22 @@ function FileManagerCollection:showCollDialog()
...
+        {}, -- separator
+        {{
+            text = _("Add all books from a folder"),
+            callback = function()
+                UIManager:close(coll_dialog)
+                self:addBooksFromFolder(true)
+            end,
+        }},
+        {{
+            text = _("Add all books from a folder and its subfolders"),
```
Comment (Frenzie) [inline frontend/apps/filemanager/filemanagercollection.lua:252]: Or "from folder and subfolders"? It's a bit long.

File: frontend/util.lua [hunk 2]
```diff
@@ -792,7 +792,7 @@ function util.findFiles(dir, cb)
             -- lfs can return nil here, as it will follow symlinks!
             local attr = lfs.attributes(path) or {}
             if attr.mode == "directory" then
-                if f ~= "." and f ~= ".." then
+                if not not_recursive and f ~= "." and f ~= ".." then
```
Comment (Frenzie) [inline frontend/util.lua:795]: Oof :-)

Comment (Frenzie) [reply frontend/util.lua:795]: Potentially more legible to do something like:

```lua
recursive = type(recursive) ~= "nil" and recursive or false
```

But of course that's uglier a few lines up.

**Review by Frenzie — APPROVED** [review]

**Comment by ertabzh** [comment]
Great new feature. But the problem is that it brings all types of files into the collection, not just "book" format files (epub, pdf...).

**Pull request merged**